cmd/kube-proxy/app/server.go

```diff
@@ -93,20 +97,48 @@ func (s *ProxyServer) Run(_ []string) error {
 	serviceConfig := config.NewServiceConfig()
 	endpointsConfig := config.NewEndpointsConfig()
 
-	protocol := iptables.ProtocolIpv4
+	protocol := utiliptables.ProtocolIpv4
 	if net.IP(s.BindAddress).To4() == nil {
-		protocol = iptables.ProtocolIpv6
+		protocol = utiliptables.ProtocolIpv6
 	}
-	loadBalancer := userspace.NewLoadBalancerRR()
-	proxier, err := userspace.NewProxier(loadBalancer, net.IP(s.BindAddress), iptables.New(exec.New(), protocol), s.PortRange)
+
+	var proxier proxy.ProxyProvider
+	var endpointsHandler config.EndpointsConfigHandler
+
+	shouldUseIptables, err := iptables.ShouldUseIptablesProxier()
 	if err != nil {
-		glog.Fatalf("Unable to create proxer: %v", err)
+		glog.Errorf("Error determining if we should use ProxierIptables: %v", err)
```

Comment:
ShouldUseIptablesProxier() guarantees false on error now, that simplifies things a lot.